docs/Implementing an ONNX backend.md

```diff
@@ -0,0 +1,122 @@
+### Implementing an ONNX backend
+
+##### What is an ONNX backend
+
+An ONNX backend is a library that can run ONNX models. As there are already many existing Deep Learning frameworks, you very likely don't need to create everything from scratch but rather create a converter that converts ONNX models to the corresponding framework specific representation and then deligate the execution to the framework. E.g. [onnx-caffe2](https://github.com/onnx/onnx-caffe2) , [onnx-coreml](https://github.com/onnx/onnx-coreml), [onnx-tensorflow](https://github.com/onnx/onnx-tensorflow) are all implemented as converters.
+
+##### Unified Backend Interface
+
+ONNX has defined an unified (Python) backend interface at https://github.com/onnx/onnx/blob/master/onnx/backend/base.py. There are three core concepts in this interface: `Device`, `Backend` and `BackendRep`.
+
+- `Device` is a lightweight abstraction of varies hardwares, e.g. CPU, CUDA gpus.
```

Comment:
of varies hardwares->over varied hardware
CUDA gpus->GPUs (not sure it's worth clarifying CUDA-ness)